sdk/storage/Azure.Storage.Blobs/src/Models/BlobProperties.cs

```diff
@@ -267,5 +267,96 @@ public BlobProperties()
             Metadata = new Dictionary<string, string>(System.StringComparer.OrdinalIgnoreCase);
             ObjectReplicationSourceProperties = new List<ObjectReplicationPolicy>();
         }
+
+        /// <summary>
+        /// Internal Constructor.
+        /// </summary>
+        internal BlobProperties(
```

Comment:
could you also check if we mutate these properties anywhere? if not then we should be good to make them read only, i.e. `{ get; }` instead of `{ get; internal set; }`

Reply:
The properties already have `{ get; internal set; }`. The only thing I found is that there are places in tests where we were using the empty constructor to do internal sets but those variables were unused in the tests, so I'm going to just remove them.

Reply:
woops misunderstood. will change it to `{get}` in a bit.